src/common/logger/logger.cpp

```diff
@@ -5,27 +5,31 @@
 #include "logger.h"
 #include <map>
 #include <spdlog/sinks/daily_file_sink.h>
-#include <spdlog\sinks\stdout_color_sinks-inl.h>
+#include <spdlog/sinks/msvc_sink.h>
+#include <spdlog/sinks/null_sink.h>
+#include <spdlog/sinks/stdout_color_sinks-inl.h>
 #include <iostream>
-#include <spdlog\sinks\null_sink.h>
 
-using namespace std;
-using namespace spdlog;
+using spdlog::sinks_init_list;
+using spdlog::level::level_enum;
+using spdlog::sinks::daily_file_sink_mt;
+using spdlog::sinks::msvc_sink_mt;
+using std::make_shared;
```

Comment:
What are the reasons for including specific functions/classes rather than the whole namespace? We are not including them in the header file so it is not namespace pollution. As for me it just slows down the coding process.

Reply:
Aside from namespace pollution in the source itself, there's also an issue of confusing functions/class names as a source tends to grow. Imagine having a source with thousands of lines full of stuff like `level_enum` - it would be very confusing to read, and having readable source is much more important than spending a some seconds on writing explicit using declarations 🙂